spec/unit/packagers/ips_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+module Omnibus
+  describe Packager::IPS do
+    let(:project) do
+      Project.new.tap do |project|
+        project.name('project')
+        project.homepage('https://example.com')
+        project.install_dir('/opt/project')
+        project.build_version('1.2.3')
+        project.build_iteration('2')
+        project.maintainer('Chef Software')
+      end
+    end
+
+    subject { described_class.new(project) }
+
+    let(:project_root) { File.join(tmp_path, 'project/root') }
+    let(:staging_dir)  { File.join(tmp_path, 'staging/dir') }
+
+    before do
+      Config.project_root(project_root)
+
+      allow(subject).to receive(:staging_dir).and_return(staging_dir)
+      create_directory(staging_dir)
+    end
+
+    it '#id is :IPS' do
+      expect(subject.id).to eq(:ips)
+    end
+
+    describe "#generate_pkg_metadata" do
+      it "should create metadata correctly" do
+        subject.generate_pkg_metadata
+        manifest_file = File.join(staging_dir, "gen.manifestfile")
+        manifest_file_contents = File.read(manifest_file)
+        expect(File.exist?(manifest_file)).to be(true)
+        expect(manifest_file_contents).to include("set name=pkg.fmri value=developer/versioning/project@2.3,2.3-2")
+        expect(manifest_file_contents).to include("set name=variant.arch value=i386")
+      end
+    end
+
+    it "should run subfunctions in the correct order during manifest creation" do
+      expect(subject).to receive(:generate_pkg_metadata).ordered
+      expect(subject).to receive(:generate_pkg_contents).ordered
+      expect(subject).to receive(:generate_pkg_deps).ordered
+      expect(subject).to receive(:check_pkg_manifest).ordered
+      subject.generate_pkg_manifest
+    end
+
+    # Bunch of methods on the subject are just calls to shellout.
+    # We are not testing the executed commands here by copy pasting them from
+    # code to here :)
```

Comment:
So the point of unit testing these methods is to ensure the commands we shell out and run are receiving the expected parameters. We should still add test coverage. We also need to add coverage for:

* `Omnibus::Packager::IPS#safe_base_package_name` (be sure to test for variations you expect to be converted...look at other packager specs for examples).
* `Omnibus::Packager::IPS#fmri_package_name`
* `Omnibus::Packager::IPS#safe_architecture`